core/src/main/scala/io/chrisdavenport/rediculous/RedisTransaction.scala

```diff
@@ -0,0 +1,90 @@
+package io.chrisdavenport.rediculous
+
+import cats._
+import cats.implicits._
+import cats.data._
+import cats.effect._
+import RedisProtocol._
+
+
+object RedisTransaction {
+
+  final case class TxQueued[A](value: RedisTx[Queued[A]])
+  object TxQueued {
+    implicit val ctx: RedisCtx[TxQueued] =  new RedisCtx[TxQueued]{
+      def run[A: RedisResult](command: NonEmptyList[String]): TxQueued[A] = TxQueued(RedisTx{for {
+        (i, base) <- State.get
+        _ <- State.set((i + 1, base ++ List(command)))
+      } yield Queued(l => RedisResult[A].decode(l(i)))})
+    }
+    implicit val applicative: Applicative[TxQueued] = new Applicative[TxQueued]{
+      def pure[A](a: A) = TxQueued(Monad[RedisTx].pure(Monad[Queued].pure(a)))
+
+      override def ap[A, B](ff: TxQueued[A => B])(fa: TxQueued[A]): TxQueued[B] =
+        TxQueued(RedisTx(
+          Nested(ff.value.value).ap(Nested(fa.value.value)).value
+        ))
+    }
+  }
+
+  final case class RedisTx[A](value: State[(Int, List[NonEmptyList[String]]), A])
+  object RedisTx {
+
+    implicit val m: Monad[RedisTx] = new StackSafeMonad[RedisTx]{
+      def pure[A](a: A): RedisTx[A] = RedisTx(Monad[State[(Int, List[NonEmptyList[String]]), *]].pure(a))
+      def flatMap[A, B](fa: RedisTx[A])(f: A => RedisTx[B]): RedisTx[B] = RedisTx(
+        fa.value.flatMap(f.andThen(_.value))
+      )
+    }
+  }
+  final case class Queued[A](f: List[Resp] => Either[Resp, A])
```

Comment:
`List` to single `A` feels weird - what's the deal there?

Reply:
The entire List of responses is returned on the `EXEC` step. So this is how we extract out our A out of the final list of responses returned. It based on the position of our command in the transaction.